tests/reftests/pin.test

```diff
@@ -1068,3 +1068,68 @@ opam-version: "2.0"
 
 No solution found, exiting
 # Return code 20 #
+### : behaviour of opam pin when confronted with a missing opam description
+### opam switch create --empty missing-opam
+### OPAMEDITOR="echo iNvAlId "
+### <pin-depends.sh>
+#!/bin/sh
+set -eu
+basedir=$(echo "$BASEDIR" | sed 's/\\/\\\\\\\\/g')
+cat << EOF >> "$1"
+pin-depends: [
+  ["$2" "file://$basedir/$3"]
+]
+EOF
+### <pin:missing-opam/pkg.opam>
+opam-version: "2.0"
+### opam pin pkg-with-typo ./missing-opam --no-action
+[NOTE] Package pkg-with-typo does not exist in opam repositories registered in the current switch.
+[pkg-with-typo.dev] synchronised (file://${BASEDIR}/missing-opam)
+[ERROR] No valid package definition found
+# Return code 5 #
+### opam pin pkg-with-typo ./missing-opam
+[NOTE] Package pkg-with-typo does not exist in opam repositories registered in the current switch.
+[pkg-with-typo.dev] synchronised (no changes)
+[ERROR] No valid package definition found
+# Return code 5 #
+### <pin:missing-opam/pkg.opam>
+opam-version: "2.0"
+### sh ./pin-depends.sh missing-opam/pkg.opam pin-pkg.dev missing-opam
+### opam pin ./missing-opam
+[NOTE] Package pkg does not exist in opam repositories registered in the current switch.
+The following additional pinnings are required by pkg.dev:
+  - pin-pkg.dev at file://${BASEDIR}/missing-opam
+Pin and install them? [Y/n] y
+[NOTE] Package pin-pkg does not exist in opam repositories registered in the current switch.
+[pin-pkg.dev] synchronised (no changes)
+[ERROR] No valid package definition found
```

Comment:
With the new behaviour, it may worth adding the package name here?

Reply:
done in follow up fixup commit